<launch>
  <!-- RealSense D456 -->
  <include file="$(find doll_head_torso)/launch/camera.launch">
    <arg name="enable_color" value="true"/>
    <arg name="enable_depth" value="false"/> <!-- pour test visuel simple -->
    <arg name="color_width"  value="640"/>
    <arg name="color_height" value="480"/>
    <arg name="color_fps"    value="30"/>
  </include>

  <!-- YOLOv8 Segmentation node -->
  <node pkg="doll_head_torso" type="yolo_seg_node.py" name="yolo_seg" output="screen">
    <param name="model_path" value="$(find doll_head_torso)/weights/best.pt"/>
    <param name="image_topic" value="/camera/color/image_raw"/>
    <param name="conf" value="0.25"/>
    <param name="iou"  value="0.5"/>
    <param name="imgsz" value="640"/>
    <param name="device" value="0"/>            <!-- 0 = GPU, 'cpu' = CPU -->
    <param name="process_every_n" value="1"/>   <!-- traite chaque frame -->
  </node>
</launch>
